src/PeerReview/ReviewRepo/Transaction.hs

```diff
@@ -1,41 +1,98 @@
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE QuasiQuotes       #-}
 module PeerReview.ReviewRepo.Transaction
-    ( saveReview
-    , findReviewsByUserId
-    , findCompleted
+    ( save
+    , findByUserId
+    , findById
+    , findByTaskId
+    , findAll
+    , update
     ) where
 
-import           Control.Monad                    (void)
 import           Data.Pool                        (Pool, withResource)
 import           Database.PostgreSQL.Simple       (Connection, Only (..),
-                                                   execute, query)
+                                                   execute, query, query_)
 import           Database.PostgreSQL.Simple.SqlQQ (sql)
 
 import           PeerReview.Types
 
 -- Save a Peer Review to db.
-saveReview :: Pool Connection -> PeerReview -> IO ()
-saveReview pool pr = withResource pool (\ conn ->
-    void $ execute conn [sql|
-        INSERT INTO peer_reviews (submission_id, task_id,comment,score,reviewer_id,status)
-        VALUES (?,?,?,?,?,?)
+save :: Pool Connection -> PeerReview -> IO PeerReviewID
+save pool pr = withResource pool (\ conn ->
+    execute conn [sql|
+        INSERT INTO peer_reviews (submission_id, task_id, submission_content, comment, score, reviewer_id, status)
+        VALUES (?,?,?,?,?,?,?)
     |] pr)
 
 -- Find all reviews for given User ID.
-findReviewsByUserId :: Pool Connection -> UserID -> IO [PeerReview]
-findReviewsByUserId pool uid = withResource pool (\ conn ->
-    query conn [sql|
-        SELECT submission_id, task_id, comment, score, reviewer_id, status
+findByUserId :: Pool Connection -> UserID -> IO [(PeerReviewID,PeerReview)]
+findByUserId pool uid = withResource pool (\ conn -> do
+    reviews <- query conn [sql|
+        SELECT id, submission_id, task_id, submission_content, comment, score, reviewer_id, status
```

Comment:
There is also `created_at` field for peer review and it's not used in the data type. We could also use `*` and then ignore the created_at in the fromRow function. I'm not sure which is better choice